internal/driver/gl/window.go

```diff
@@ -220,14 +220,12 @@ func (w *window) SetPadded(padded bool) {
 
 	w.canvas.content.Move(w.contentPos())
 
-	w.resizeToContent()
+	runOnMain(w.resizeToContent)
 }
 
 func (w *window) resizeToContent() {
-	runOnMain(func() {
-		w.fitContent()
-		w.viewport.SetSize(w.screenSize(w.Canvas().Content().Size()))
-	})
+	w.fitContent()
+	w.viewport.SetSize(w.screenSize(w.Canvas().Content().Size()))
```

Comment:
This must be run on the main thread. If we remove the safety within the method then it could be called incorrectly again.
I know you have reason for the change - but I'm not sure what?

Reply:
The loop already runs on the main thread and leaving the guard in the method didn't work.
I didn't figure out why.